lanternmq/rabbitmq/messagequeue.go

```diff
@@ -0,0 +1,346 @@
+type MessageQueue struct {
+	connection *amqp.Connection
+	channels   []*amqp.Channel
+}
+
+// Messages wraps the delivery channel.
+type Messages struct {
+	deliveryChannel <-chan amqp.Delivery
+}
+
+// addChannel adds the given channel to the MessageQueue.channels array and returns the
+// index to that array casted to a lanternmq.ChannelID.
+func (mq *MessageQueue) addChannel(ch *amqp.Channel) (lanternmq.ChannelID, error) {
+	if mq.channels == nil {
+		mq.channels = []*amqp.Channel{}
+	}
+	mq.channels = append(mq.channels, ch)
+	index := len(mq.channels) - 1
+	id := lanternmq.ChannelID(index)
+
+	return id, nil
+}
+
+// getChannel retrieves the channel provided by `id` by casting `id` back to an integer and
+// retrieving the channel at the corresponding index of MessageQueue.channels array.
+func (mq *MessageQueue) getChannel(id lanternmq.ChannelID) (*amqp.Channel, error) {
+	idInt, ok := id.(int)
+	if !ok {
+		return nil, errors.New("ChannelID not of correct type")
+	}
+	if idInt >= len(mq.channels) {
+		return nil, errors.New("no channel with the requested ID was found")
+	}
+	ch := mq.channels[idInt]
+	return ch, nil
+}
+
+// Connect creates a connection to a RabbitMQ service.
+func (mq *MessageQueue) Connect(username string, password string, host string, port string) error {
+	s := fmt.Sprintf("amqp://%s:%s@%s:%s/", username, password, host, port)
+	conn, err := amqp.Dial(s)
+	if err != nil {
+		err = errors.New("unable to connect to message queue")
+	}
+	mq.connection = conn
+
+	return err
+}
+
+// CreateChannel creates a channel to the RabbitMQ service that has already been connected to.
+// If the RabbitMQ service has not been connected to already, an error is thrown.
+// The channel's ID is returned.
+func (mq *MessageQueue) CreateChannel() (lanternmq.ChannelID, error) {
+	var err error
+	if mq.connection == nil {
+		err = errors.New("connection must exist before creating a channel")
+		return "", err
+	}
+	ch, err := mq.connection.Channel()
+	if err != nil {
+		err = errors.New("unable to create channel")
+		return "", err
+	}
+	id, err := mq.addChannel(ch)
+
+	return id, err
+}
+
+// NumConcurrentMsgs defines how many messages the user can process from the channel at one time.
+func (mq *MessageQueue) NumConcurrentMsgs(chID lanternmq.ChannelID, num int) error {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return err
+	}
+
+	err = ch.Qos(
+		num,   // prefetch count
+		0,     // prefetch size
+		false, // global
+	)
+	if err != nil {
+		err = errors.New("unable to set the number of concurrent messages that can be handled")
+		return err
+	}
+	return err
+}
+
+// DeclareQueue creates a queue with the given name on the given channel using RabbitMQ's
+// QueueDeclare method with the following arguments:
+// * name: qName
+// * durable: true
+// * autoDelete: false
+// * exclusive: false
+// * noWait: false
+// * args: nil
+func (mq *MessageQueue) DeclareQueue(chID lanternmq.ChannelID, qName string) error {
```

Comment:
Consider using a parameter struct as an argument to provide the desired queue settings.  This makes the interface/implementation more applicable to a wider range of scenarios.